opm/common/OpmLog/LogBackend.hpp

```diff
@@ -75,6 +79,7 @@ namespace Opm
         int64_t m_mask;
         std::shared_ptr<MessageFormatterInterface> m_formatter;
         std::shared_ptr<MessageLimiter> m_limiter;
+        int64_t m_level;
```

Comment:
It's because it uses the same type as the "message type" flags, and there is some reason to need to combine them for filtering in the Logger class so it cannot be an enum class.

I think the idea here is good, but there is some overlap with the enabled message concept in the Logger class so perhaps we should refactor that instead?